### ISSUE TITLE
Acceleration structures

### ISSUE THREAD

**Issue by w23** [tsoding/ray-tracer#32]
For triangle objects (or sufficiently complex scenes made of balls, boxes and walls) you will notice that iterating through all objects is slow. Skip objects that will not be hit by current ray (or ray sample). Hints: BVH, BSP, kd-trees, ...

**Comment by rexim** [comment]
@w23 yeah, I was thinking about using those in the future. Thanks for filing an issue for that! :)
